### PR TITLE
feat: Add support for Private Service Connect

### LINKED ISSUE #389: Support Private Service Connect configuration
Add a flag for PSC similar to the `PrivateIP` flag (see internal/api/v1/authproxyworkload_types.go:323)

Depends on a release of the proxy with PSC support.

### PR DESCRIPTION
**Pull request by hessjcg**
Adds support to configure private service connect through the operator. This will set the proxy's "psc=true" parameter
to the instance connection string passed to the proxy.

In accordance with the validation in the proxy, Users may not set both PrivateIP and PSC at the same time. See [root.go:517](https://github.com/hessjcg/cloudsql-proxy/blob/e0a97ddd5bea94054b1da0c3f0844ab47ad6f126/cmd/root.go#L517)

Fixes #389

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ _Appears in:_
 | `port` _integer_ | Port (optional) sets the tcp port for this instance. If not set, a value will be automatically assigned by the operator and set as an environment variable on all containers in the workload named according to PortEnvName. The operator will choose a port so that it does not conflict with other ports on the workload. |
 | `autoIAMAuthN` _boolean_ | AutoIAMAuthN (optional) Enables IAM Authentication for this instance. Default value is false. |
 | `privateIP` _boolean_ | PrivateIP (optional) Enable connection to the Cloud SQL instance's private ip for this instance. Default value is false. |
+| `psc` _boolean_ | PSC (optional) Enable connection to the Cloud SQL instance's private service connect for this instance. May not be used with PrivateIP. Default value is false. |
```
Comment (jackwotherspoon) [inline docs/api.md:108]: same here

File: internal/api/v1/authproxyworkload_types.go [hunk 2]
```diff
@@ -322,6 +322,12 @@ type InstanceSpec struct {
 	//+kubebuilder:validation:Optional
 	PrivateIP *bool `json:"privateIP,omitempty"`
 
+	// PSC (optional) Enable connection to the Cloud SQL instance's private
+	// service connect for this instance. May not be used with PrivateIP.
```
Comment (jackwotherspoon) [inline internal/api/v1/authproxyworkload_types.go:326]: and here

Comment (hessjcg) [reply internal/api/v1/authproxyworkload_types.go:326]: Fixed.

File: config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml [hunk 3]
```diff
@@ -994,6 +994,9 @@ spec:
                       privateIP:
                         description: PrivateIP (optional) Enable connection to the Cloud SQL instance's private ip for this instance. Default value is false.
                         type: boolean
+                      psc:
+                        description: PSC (optional) Enable connection to the Cloud SQL instance's private service connect endpoint for this instance. May not be used with PrivateIP. Default value is false.
```
Comment (jackwotherspoon) [inline config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml:998]: i still find this read's weird. Do you? If you are saying `connection to the Cloud SQL instance's...` then I don't think you need to also have `for this instance`. It is repetitive and hinders the readability for me.
```suggestion
                        description: PSC (optional) Enable connection to the Cloud SQL instance's private service connect endpoint. May not be used with PrivateIP. Default value is false.
```

Comment (jackwotherspoon) [reply config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml:998]: same in below places as well

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]

File: config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml (hunk 3 above)
Comment (hessjcg) [reply config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml]: Yeah, that's better. I've updated accordingly. 
Btw, these files are generated from the comments in authproxworkload_types.go by the Makefile using controller-runtime tools.

**Pull request merged**